### ISSUE TITLE
`hab unload` followed by `hab start` on a transient service can result in erroneous spec file deletion

### ISSUE THREAD

**Issue by stevendanna** [habitat-sh/habitat#4806]
`hab unload` immediately followed by `hab start` can result in the running service no longer having an on disk spec file.

## Reproduction

1) Start a service:

```
# hab sup start chef/authz-service --bind automate-postgresql:automate-postgresql.default
hab-sup(MN): Supervisor starting chef/authz-service. See the Supervisor output for more details.
# ls /hab/sup/default/specs/authz-service.spec
/hab/sup/default/specs/authz-service.spec
```

2) Unload & Start the service.  Ensure the start would produce a different spec file (here I've added `--channel`):

```
# hab sup unload chef/authz-service; hab sup start chef/authz-service --bind automate-postgresql:automate-postgresql.default --channel dev
# ls /hab/sup/default/specs/authz-service.spec
ls: cannot access '/hab/sup/default/specs/authz-service.spec': No such file or directory
```

## Impact
While the spec file does not exist, the service is running:

```
root@automate-deployment:/opt/automate-deployment# ps auxww | grep authz-service
hab      10374  0.1  0.7  28816 14912 ?        Sl   09:27   0:00 authz-service /hab/svc/authz-service/config/config.yml
```

The `hab` command reports different status information for this service depending on the command used:

```
# hab sup status | grep authz-service
chef/authz-service/0.1.0/20180326071234            standalone  up     106         10374  authz-service.default            transient
# hab sup status chef/authz-service
chef/authz-service is not currently loaded.
```

Unload becomes useless as there is no spec file for it to delete:

```
# hab sup unload chef/authz-service
# hab sup status | grep authz
chef/authz-service/0.1.0/20180326071234            standalone  up     224         11592  authz-service.default            transient
```

## Likely Cause

Ultimately, I think the cause is this:

- If the unload and start happen quickly enough, both the removal and rewrite of the spec file will occur within in the interval of the file watcher.  This will result in the file watcher detecting the operation as a spec change: https://github.com/habitat-sh/habitat/blob/master/components/sup/src/manager/spec_watcher.rs#L206-L208

- When the RemoveService event is handled, it deletes the spec file:

https://github.com/habitat-sh/habitat/blob/master/components/sup/src/manager/mod.rs#L857-L861

## Related Issues

- Editing a spec file results in the spec file being removed and eventually the service stopping:

```
# grep channel /hab/sup/default/specs/authz-service.spec
channel = "dev"
# sed -i '/channel/d' /hab/sup/default/specs/authz-service.spec
# ls /hab/sup/default/specs/authz-service.spec
ls: cannot access '/hab/sup/default/specs/authz-service.spec': No such file or directory		
```

- Running `hab unload` and `hab start` quickly produces no change. (https://github.com/habitat-sh/habitat/issues/4807)

**Comment by stevendanna** [comment]
I think this bug is fixed with hab-sup 0.56.0

**Comment by reset** [comment]
This is at the very least no longer relevant since transient / persistent services are no longer a thing 😄
